### ISSUE TITLE
FEAT: npm version support running via systemd socket activation

### ISSUE THREAD

**Issue by dinkonin** [dbgate/dbgate#834]
It would be nice to have the ability to run the NPM version of dbgate via systemd socket activation.
I tried to do a basic socket activation like this:

```ini
[Unit]
Description=Socket for Dbgate Node.js Application

[Socket]
ListenStream=3636
Accept=false

[Install]
WantedBy=sockets.target

[Unit]
Description=DbGate Web Service
```

```ini
[Service]
User=dbgate
Group=dbgate
Environment="PORT=3636"
ExecStart=/usr/bin/dbgate-serve

[Install]
WantedBy=multi-user.target

```

The socket is is working but when dbgate activates it begins listiening on a random port.

Maybe this would help. https://github.com/derhuerst/node-systemd

**Comment by ProjectInfinity** [comment]
Am I right to assume that the issue you are facing is that dbgate listens to a random port instead of the provided 3636 passed in the environment variable?

**Comment by dinkonin** [comment]
Yes, IMO it detects that the port is open by the systemd socket.
--
*Kostadin Ninev*
